### ISSUE TITLE
early.stop.round not working in R?

### ISSUE THREAD

**Issue by SirSpunky** [dmlc/xgboost#982]
Am I dumb or is early.stop.round not working in R?

> clf <- xgb.cv(      params              = param, 
                     data                = train_matrix_xgb, 
                     nrounds             = 1000,
                     verbose             = 1,
                     nfold               = 5,
                     early.stop.round    = 10,
                     maximize            = FALSE
 )
[0]	train-auc:0.797073+0.033895	test-auc:0.783956+0.041039
[1]	train-auc:0.800843+0.036190	test-auc:0.786327+0.043101
[2]	train-auc:0.815333+0.010959	test-auc:0.802576+0.017459
[3]	train-auc:0.819934+0.007955	test-auc:0.808093+0.013064
[4]	train-auc:0.824674+0.006169	test-auc:0.811653+0.011376
[5]	train-auc:0.827333+0.004562	test-auc:0.813708+0.010619
[6]	train-auc:0.829889+0.003716	test-auc:0.816235+0.010792
[7]	train-auc:0.830157+0.003157	test-auc:0.817124+0.010752
[8]	train-auc:0.831281+0.002608	test-auc:0.817517+0.010698
[9]	train-auc:0.833064+0.001278	test-auc:0.818807+0.010039
[10]	train-auc:0.832605+0.002623	test-auc:0.818192+0.011359
Stopping. Best iteration: 1

In the manual it says "If set to an integer k, training with a validation set will stop if the performance keeps getting worse consecutively for k rounds." However, now it stops after the first 10 rounds, even if test-auc is still rising.

Could this be a bug? I'm using the latest version of R (and RStudio) in Windows 10 and the latest xgboost installed from CRAN (v0.4-3).

**Comment by chibberto** [comment]
Set maximize = TRUE?

**Comment by SirSpunky** [comment]
Ahhh, I knew I was missing something. Thanks a lot!
